internal/vehicle_template.go

```diff
@@ -28,11 +29,22 @@ func NewVehicleTemplates(logger zerolog.Logger, vsd gateways.VehicleSignalDecodi
 
 // GetTemplateSettings checks for any new template settings and if so updates the local settings, returning the latest
 // settings. Logs if encounters errors along the way. Continues and gets local settings if can't get anything from remote. Errors if can't get anything useful.
-func (vt *vehicleTemplates) GetTemplateSettings(addr *common.Address) (*models.TemplatePIDs, *models.TemplateDeviceSettings, error) {
+func (vt *vehicleTemplates) GetTemplateSettings(addr *common.Address, vl loggers.VINLogger, unitID uuid.UUID) (*models.TemplatePIDs, *models.TemplateDeviceSettings, error) {
 	vinConfig, err := vt.lss.ReadVINConfig() // should this be more of VIN info? stored VIN info
 	if err != nil {
 		vt.logger.Err(err).Msg("could not read local settings for stored VIN, continuing")
 	}
+
+	// if we can't find VIN from template file, then we need to query it from OBD
+	vinResp, err := vl.GetVIN(unitID, nil)
+	if err != nil {
+		vt.logger.Err(err).Msg("Unable to get VIN from vinLogger, continuing")
+	}
+	if vinResp != nil {
+		vinConfig = &models.VINLoggerSettings{}
```

Comment:
if this is succesful, we may be able to to `vt.lss.SaveVINConfig()` or something like that to persist for future session

Reply:
I tried to save the Vin using `vt.lss.WriteVINConfig(*vinConfig)`, but it caused panic afterward during fingerprint. If we persist it here, "vin_query_name" would be saved to the file as "", which is causing problems, I guess. Thoughts?

Reply:
hmm probably stuff we haven't tested much. Does the fingerprint save the file? Maybe just handling it from fingerprint and have this pick it up if the file exists.

Reply:
Also, the `/tmp/vin-settings.json` will be written by downstream anyway, so when the device wakes up, the file will be there already. This is my understanding.

Reply:
sounds good, let's do an experiment to validate that this is true - I wrote that code but was never too sure about it

Reply:
Verified